HHH-18181 Include property and class for PropertyBinder errors

schosin opened the pull request:
https://hibernate.atlassian.net/browse/HHH-18181

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

gavinking commented on hibernate-core/src/main/java/org/hibernate/boot/model/internal/PropertyBinder.java:
```diff
@@ -292,8 +292,9 @@ private void callAttributeBinders(Property property, Map<String, PersistentClass
-				throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "'", e );
+			catch (Exception e) {
+				throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "' for property '"
+						+ property.getName() + "' in class '" + holder.getEntityName() + "'", e );
```
```suggestion
						+ property.getName() + "' of class '" + holder.getEntityName() + "'", e );
```

gavinking commented on hibernate-core/src/main/java/org/hibernate/boot/model/internal/PropertyBinder.java:
```diff
@@ -292,8 +292,9 @@ private void callAttributeBinders(Property property, Map<String, PersistentClass
-			catch ( Exception e ) {
-				throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "'", e );
+			catch (Exception e) {
+				throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "' for property '"
```
```suggestion
				throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "' for attribute '"
```

schosin replied:
I can also squash the commits if preferred.

beikov commented on hibernate-core/src/main/java/org/hibernate/boot/model/internal/PropertyBinder.java:
```diff
@@ -292,8 +292,9 @@ private void callAttributeBinders(Property property, Map<String, PersistentClass
-			catch ( Exception e ) {
-				throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "'", e );
+			catch (Exception e) {
+				throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "' for attribute '"
+						+ property.getName() + "' of class '" + holder.getEntityName() + "'", e );
```
Why not this?

```suggestion
				throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "' for attribute " + qualify( holder.getPath(), name ), e );
```

schosin replied:
Did not know / notice that usage, that is all. Should I change it? Most other usages of qualify in that class use property instead of attribute, should I change the wording for my change back again as well?

```java
throw new AnnotationException( "error processing @AttributeBinderType annotation '" + containingAnnotation + "' for property '"
		+ qualify( holder.getPath(), name ) + "'", e );
```

sebersole replied:
> Should I change it? 

Yes.

> Most other usages of qualify in that class use property instead of attribute, should I change the wording for my change back again as well?

Yes, that seems reasonable.  If you are willing.  I agree with @gavinking  in that I prefer `attribute` - its the phrase JPA uses.  But I'm fine with either; it should just be consistent.

sebersole commented:
@schosin - We are investigating a move to the Apache Software License v2. If you agree with allowing this change to be relicensed under that license, please add this text to the subject -

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

sebersole commented:
@schosin - did you want to apply the suggestions and accept the possible license change?

schosin commented:
I already accepted an e-mail regarding the license change. I added the text to the PR, not sure if that's correct location. I made the change including the use of `property`, rebased on main and will force push the branch once I've successfully ran the build.

sebersole commented:
> I already accepted an e-mail regarding the license change.

Ah, sorry about that.  The processes aren't really integrated.

> I added the text to the PR

Thanks!

> get to that either later today or on the weekend

No problem.

schosin commented:
Okay, I actually rebased now and was able to run `./gradlew :hibernate-core:build` with one test failure in TypesafeNamedQueryTest that does not seem to be related to these changes. `./gradlew clean build` sadly didn't complete since BasicBlobTest in envers does not like windows paths: `Illegal char <:> at index 2: /C:/Users/...`
